.github/workflows/cre-system-tests.yaml

```
@@ -204,13 +222,31 @@ jobs:
         run: |
           # Sleep to allow testcontainers to stop
           # Before running it with flakeguard add this bit back: { go test... -json; exit_code=$?; ../../tools/ci/wait-for-containers-to-stop.sh 30; exit $exit_code; }
-          go test github.com/smartcontractkit/chainlink/system-tests/tests/smoke/cre -v -run "^(${TEST_NAME})$" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 -json 2>&1 | tee /tmp/gotest.log | gotestloghelper -ci -singlepackage -hidepassingtests=false
+          go test github.com/smartcontractkit/chainlink/system-tests/tests/smoke/cre -v -run "^(${TEST_NAME})$" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 2>&1 | tee /tmp/gotest.log
+
+      - name: Run CRE Regression system tests
+        if: ${{ inputs.with_regression == 'true' }}
+        shell: bash
+        working-directory: system-tests/tests
+        env:
+          TEST_NAME: ${{ matrix.tests.test_name }}
+          # override chip ingress image to use remote image. We have added this env var here, instead of the "Start local CRE" step, because
+          # Beholder stack will be started only for the Beholder test
+          CHIP_INGRESS_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/atlas-chip-ingress:bbac3c825b061546980fa9d7dc0f3e8c34347bcf
+          BILLING_PLATFORM_SERVICE_IMAGE: ${{ secrets.AWS_ACCOUNT_ID_PROD }}.dkr.ecr.${{ secrets.QA_AWS_REGION }}.amazonaws.com/billing-platform-service:v1.36.2-amd64
+          TEST_TIMEOUT: 200m
+        run: |
+          # Sleep to allow testcontainers to stop
+          # Before running it with flakeguard add this bit back: { go test... -json; exit_code=$?; ../../tools/ci/wait-for-containers-to-stop.sh 30; exit $exit_code; }
+          go test github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre -v -run "^(${TEST_NAME})$" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 2>&1 | tee /tmp/gotest.log
```

Pre-existing, but just noticed: Given that the [docs](https://pkg.go.dev/cmd/go/internal/test) for `go test -list` say:
```
-list regexp
	    List tests, benchmarks, fuzz tests, or examples matching the regular
	    expression. No tests, benchmarks, fuzz tests, or examples will be run.
	    **This will only list top-level tests. No subtest or subbenchmarks will be
	    shown.**
```
Should we drop the trailing `$` from our regexes? Otherwise sub-tests are excluded.
```suggestion
          go test github.com/smartcontractkit/chainlink/system-tests/tests/regression/cre -v -run "^(${TEST_NAME})" -timeout ${TEST_TIMEOUT} -count=1 -test.parallel=1 2>&1 | tee /tmp/gotest.log
```

---

Thanks, @jmank88, updated.

---

Retracting my previous statement, the way the test matrix is defined requires ^$ to wrap the test name; otherwise, some tests will be skipped.

---

I think we must be miscommunicating. Including `$` will skip sub-tests. What effect do you see when it is excluded? You should only see more test names match, never fewer.

---

We use `t.Run` for nested tests, which are never excluded, so all the tests are eventually correctly parsed and executed compared to `$`